codec/unknownproto/regression_test.go

```diff
@@ -19,8 +18,6 @@ func TestBadBytesPassedIntoDecoder(t *testing.T) {
 	decoder := cfg.TxConfig.TxDecoder()
 	tx, err := decoder(data)
 
-	// TODO: When issue https://github.com/cosmos/cosmos-sdk/issues/7846
-	// is addressed, we'll remove this .Contains check.
-	require.Contains(t, err.Error(), io.ErrUnexpectedEOF.Error())
```

Comment:
I don't think this is correct, the comment means to say that once we get a specific error we can compare against it, not make the error check more generic.